### ISSUE TITLE
Nicer errors on requests?

### ISSUE THREAD

**Issue by brandondube** [goji/goji#64]
Suppose you have an endpoint `pat.Get("/foo")` and the user/client does an `http.Post(".../foo")`.  They will get a 404 not found, but they "really" should get a 400/bad method.  Is there a way to make Goji produce "dummy" routes that respond with bad method for every HTTP verb that isn't implemented for a particular endpoint on a mux?

**Comment by zenazn** [comment]
Yeah, that's definitely an outcome some folks may prefer. It's not a core feature of Goji today, and it's hard to implement on top of the current `Pattern` interface for the general case.

Funnily enough, the older version of Goji (`zenazn/goji`) supports something like this for OPTIONS calls. Might be worth a look if this feature is very important to you!
